custom/ilsgateway/api.py

```diff
@@ -275,6 +295,25 @@ def sms_user_sync(self, ilsgateway_smsuser, **kwargs):
         return sms_user
 
     def location_sync(self, ilsgateway_location):
+        def get_or_create_msd_zone(region):
+            msd_name = _get_msd_name(region.name)
+            msd_code = MSDZONE_MAP[msd_name][0]
+            try:
+                sql_msd_loc = SQLLocation.objects.get(
+                    domain=self.domain,
+                    site_code=msd_code
+                )
+                msd_location = Loc.get(sql_msd_loc.location_id)
+            except SQLLocation.DoesNotExist:
+                msd_location = Loc(parent=loc_parent)
+
+            msd_location.domain = self.domain
+            msd_location.name = msd_name
+            msd_location.location_type = 'MSDZONE'
+            msd_location.site_code = MSDZONE_MAP[msd_name][0]
+            msd_location.save()
```

Comment:
it might be worth adding logic to only save the zone if there are changes. though I know the number of regions is not so high so probably not worth blocking on it.